src/renderer/components/dock/log-tab.store.ts

```diff
@@ -167,18 +173,34 @@ export class LogTabStore extends DockTabStore<LogTabData> {
 
     if (error) {
       logger.warn(`[LOG-TAB-STORE]: data for ${tabId} was invalid`, error);
-      this.closeTab(tabId);
+      runInAction(() => {
+        this.closeTab(tabId);
+      });
 
       return undefined;
     }
 
     return value;
   }
 
+  @action
   public closeTab(tabId: string) {
     this.clearData(tabId);
     this.dockManager.closeTab(tabId);
   }
+
+  /**
+   * Get a set of namespaces which pod tabs care about
+   */
+  public getNamespaces(): string[] {
```

Comment:
I don't get why we should get namespaces in some way and watch them in `logs.tsx`. The pod which logs we're looking at (as well as it's "sibling" pods with same owner) could have only 1 namespace. So why we need to collect more than one? Isn't it enough to pick `namespace` attribute from the previewed pod like `pod.getNs()`?

Reply:
It is, however, this gets the namespaces for all tabs not just the currently open one. 

This is needed for when we reopen Lens and the `pods` view might not be open yet. So the Logs component needs to register a watch for pods in all of these namespaces.

I guess I could watch pods in a single namespace for each tab. WDYT?